### ISSUE TITLE
AppShell is not re-rendering when $page from "$app/stores" changes

### ISSUE THREAD

**Issue by Angel-Ponce** [svelteuidev/svelteui#439]
### What package has an issue

@svelteuidev/core

### A clear and concise description of what the bug is

Simple example: 

```svelte
<script>
    import { page } from "$app/stores"
</script>

{$page.route.id} <<<<---- this line change when route change

<AppShell>
 <Header zIndex={101} slot="header" height={60} fixed>
  your header content
  {$page.route.id} <<<---- this line is not changing when route change
 </Header>
 <Navbar
 hiddenBreakpoint={'sm'}
 slot="navbar"
 fixed
 >
  your navbar content
 </Navbar>
 <ShellSection grow>
  <slot />
 </ShellSection>
</AppShell>
```

### Steps To Reproduce

1. Import `page` from "$app/stores"
2. Use the following property of `page` : `$page.route.id` inside `<AppShell />` (Preffer use inside `<Header />` or `<Navbar />` sections) 
3. Make two differents routes
4. Navigate between both routes
5. The `$page.route.id` will not change in <AppShell />

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

### Relevant Assets

https://github.com/svelteuidev/svelteui/assets/60164099/63a7c770-3053-4928-8b60-46feb5610bac

(Watch the behaviour of the red and green boxes when the route changes)

**Comment by Angel-Ponce** [comment]
**!Update**
I found a temporal fix to ignore this problem:

Use the following svelte directive:

```svelte
{#key $page.url.pathname} <--- using the `key` directive to fix the bad behaviour with `AppShell`
    <AppShell>
            <Header zIndex={101} slot="header" height={60} fixed>
             ...
     </AppShell>
{/key}

**Comment by notramo** [comment]
This workaround throws away the advantages of SvelteKit client-side routing, causing the layout to be re-rendered.

**Comment by Angel-Ponce** [comment]
Yep, but it's only a work around if you need to use this feature with the router.

**Comment by Angel-Ponce** [comment]
What about the state of this repo?? It is currently abandoned?

**Comment by BeeMargarida** [comment]
> What about the state of this repo?? Is it currently abandoned?

No, we are working on fixing the lib and making it compatible with svelte5

**Comment by Angel-Ponce** [comment]
> > What about the state of this repo?? Is it currently abandoned?
> 
> No, we are working on fixing the lib and making it compatible with svelte5

Thanks!! 💯

**Comment by Angel-Ponce** [comment]
Any update?
